include/swift/SIL/SILOpenedArchetypesTracker.h

```diff
@@ -20,6 +20,12 @@
 
 namespace swift {
 
+// Disable MSVC warning: multiple copy constructors specified.
+#if defined(_MSC_VER)
+#pragma warning(push)
+#pragma warning(disable: 4521)
+#endif
```

Comment:
Is the warning legitimate, or an MSVC bug? Can we slap an `enable_if` on a generic constructor somewhere to avoid formally having multiple copy ctors?